xdem/spstats.py

```diff
@@ -9,19 +9,92 @@
 import random
 import warnings
 from functools import partial
-from typing import Callable, Union
+from typing import Callable, Union, Optional
 
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 from scipy import integrate
 from scipy.optimize import curve_fit
+from skimage.draw import disk
+from scipy.interpolate import RegularGridInterpolator, LinearNDInterpolator, griddata
+from xdem.spatial_tools import nmad
 
 with warnings.catch_warnings():
     warnings.filterwarnings("ignore", category=DeprecationWarning)
     import skgstat as skg
     from skgstat import models
 
+def robust_interp_nd_binning(df: pd.DataFrame, var_names: list[str], statistic : Union[str, Callable, None] = nmad, min_count: Optional[int] = 100) -> Callable:
+    """
+    Estimate an interpolated function for an N-dimensional binning.
+    Relies on scipy.ndimage.map_coordinates: see for more information on input arguments.
+
+    :param df: dataframe of binned values according to explanatory variables
+    :param var_names: list of variable names to use
```

Comment:
So `var_names` are the "X" variables, and `statistic` is the "y" variable?

In scikit-learn, the associated terminology would be:
```python
"""
:param X: The training data column(s).
:param y: The target data column.
"""
```
In scipy and numpy, it's `x` and `y`.

I think `var_names` makes sense, but perhaps the `statistic` argument could be `target` (or something similar) instead to be more consistent with other tools?

Reply:
It's a bit different here as we are doing binned statistics (not regression), so I try to stay consistent with `scipy.binned_statistic_dd` which uses the terminology `samples`, `values` and `statistic`.

Reply:
Ah right.. But still, the "statistic" is the target variable for the interpolation. When I started reading, I thought "statistic" was the error function to use in the error minimization behind the scenes!

Reply:
I'm sure this confusion can be circumvented either by renaming or by elaborating a bit on the documentation.